cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/MermaidHelper.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2024, Fraunhofer AISEC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *                    $$$$$$\  $$$$$$$\   $$$$$$\
+ *                   $$  __$$\ $$  __$$\ $$  __$$\
+ *                   $$ /  \__|$$ |  $$ |$$ /  \__|
+ *                   $$ |      $$$$$$$  |$$ |$$$$\
+ *                   $$ |      $$  ____/ $$ |\_$$ |
+ *                   $$ |  $$\ $$ |      $$ |  $$ |
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg
+
+import de.fraunhofer.aisec.cpg.graph.Node
+import de.fraunhofer.aisec.cpg.passes.Pass
+import de.fraunhofer.aisec.cpg.passes.hardDependencies
+import de.fraunhofer.aisec.cpg.passes.hardExecuteBefore
+import de.fraunhofer.aisec.cpg.passes.isFirstPass
+import de.fraunhofer.aisec.cpg.passes.isLastPass
+import de.fraunhofer.aisec.cpg.passes.softDependencies
+import de.fraunhofer.aisec.cpg.passes.softExecuteBefore
+import kotlin.reflect.KClass
+
+/** Helper to create a mermaid graph given a list of passes. */
+object MermaidHelper {
+    /** Helper function to replace the first and last passes names by their identifier. */
+    private const val UNKNOWNPASS = "UnknownPass"
+    private const val FIRSTPASSESSUBGRAPHIDENTIFIER = "FirstPassesSubgraph"
```

Comment:
the canonical naming would probably be `FIRST_PASSES_SUBGRAPH_IDENTIFIER` or something similar.